api/routes_test.go

```diff
@@ -0,0 +1,13 @@
+package api
+
+import (
+	"testing"
+)
+
+func TestBuildHttpRoutes(t *testing.T) {
```

Comment:
All functions, including test functions, need a docstring. This one seems a bit odd to me, because I could imagine adjusting the api object and buildRoutes method such that just calling buildHttpRoutes would cause a panic. The full set of routes get built every time you call `newServerTester`, so this might be a situation where we don't need this test.

Reply:
doc strings and tests should come after design decisions. i don't know where `api.router =` and cleanCloseHandler etc should live so I don't know how to document / improve this test. i agree it may likely be removed. so my suggestion is the lead core devs take this over and arbitrate.

Reply:
I think `api.router =` and `cleanCloseHandler` etc should live in `api.go`, because it's where the API is initialized. `routes.go` constructs the router, `api.go` constructs the API object.